phpstan-baseline.neon

```diff
@@ -0,0 +1,66 @@
+parameters:
+	ignoreErrors:
+		-
+			message: "#^Access to constant SPAN_KIND_INTERNAL on an unknown class Opentelemetry\\\\Proto\\\\Trace\\\\V1\\\\Span\\\\SpanKind\\.$#"
```

Comment:
I wonder why PHPStan could not recognise this :thinking: isn't this part of the dependencies?

Reply:
open-telemetry/sdk is not in the dependencies list, it's an optional dependency. Plugin's code for autoinstrumentation initialization checks if SDK is available
https://github.com/shopware/opentelemetry/blob/main/_register.php#L12

Newly added metrics implementation also does not require sdk, if it's absent - open-telemetry/api will just use noop meter.

Solution here could be including open-telemetry/sdk at least as dev dependency [(similar problem discussed on reddit)](https://www.reddit.com/r/PHP/)

Reply:
sounds like a reasonable solution :+1:

Reply:
It is part of our requirement as we require `ext-opentelemetry`. Make phpstan does not understand the classes of that extension.

We added the if condition when maybe the build environment had that extension, but not the runtime and not to explode the whole thing

Reply:
Fixed. Seems there was a dependency on the files that exist only in https://github.com/open-telemetry/opentelemetry-php. Replaced with the files from open-telemetry/api package.